File: src/main/java/net/lomeli/lomlib/client/addon/jei/crafting/ShapedFluidRecipeHandler.java
package net.lomeli.lomlib.client.addon.jei.crafting;

import mezz.jei.api.recipe.IRecipeHandler;
import mezz.jei.api.recipe.IRecipeWrapper;

import javax.annotation.Nonnull;

import java.util.List;

import net.lomeli.lomlib.client.addon.jei.LomLibPlugin;
import net.lomeli.lomlib.core.recipes.ShapedFluidRecipe;

public class ShapedFluidRecipeHandler implements IRecipeHandler<ShapedFluidRecipe> {
    @Nonnull
    @Override
    public Class<ShapedFluidRecipe> getRecipeClass() {
        return ShapedFluidRecipe.class;
    }

    @Nonnull
    @Override
    public String getRecipeCategoryUid() {
        return LomLibPlugin.CRAFTING;
    }

    @Nonnull
    @Override
    public IRecipeWrapper getRecipeWrapper(@Nonnull ShapedFluidRecipe recipe) {
        return new ShapedFluidRecipeWrapper(recipe);
    }

    @Override
    public boolean isRecipeValid(@Nonnull ShapedFluidRecipe recipe) {
        if (recipe.getRecipeOutput() == null)
            return false;
        int inputCount = 0;
        for (Object input : recipe.getInput()) {
            if (input instanceof List) {
                if (((List) input).size() == 0)
                    return false;
            }
            if (input != null)
                inputCount++;
        }
        return inputCount > 0;
    }

    @Nonnull
    @Override
    public String getRecipeCategoryUid(@Nonnull ShapedFluidRecipe recipe) {
        return LomLibPlugin.CRAFTING;
    }
}

File: src/main/kotlin/net/lomeli/lomlib/client/addon/jei/crafting/ShapedFluidRecipeHandler.kt
package net.lomeli.lomlib.client.addon.jei.crafting

import mezz.jei.api.recipe.IRecipeHandler
import mezz.jei.api.recipe.IRecipeWrapper

import net.lomeli.lomlib.client.addon.jei.LomLibPlugin
import net.lomeli.lomlib.core.recipes.ShapedFluidRecipe

class ShapedFluidRecipeHandler : IRecipeHandler<ShapedFluidRecipe> {
    override fun getRecipeClass(): Class<ShapedFluidRecipe> {
        return ShapedFluidRecipe::class.java
    }

    override fun getRecipeCategoryUid(): String {
        return LomLibPlugin.CRAFTING
    }

    override fun getRecipeWrapper(recipe: ShapedFluidRecipe): IRecipeWrapper {
        return ShapedFluidRecipeWrapper(recipe)
    }

    override fun isRecipeValid(recipe: ShapedFluidRecipe): Boolean {
        if (recipe.recipeOutput == null)
            return false
        var inputCount = 0
        if (recipe.getInput() != null) {
            for (input in recipe.getInput()) {
                if (input is List<*>) {
                    if (input.size == 0)
                        return false
                }
                if (input != null)
                    inputCount++
            }
        }
        return inputCount > 0
    }

    override fun getRecipeCategoryUid(recipe: ShapedFluidRecipe): String {
        return LomLibPlugin.CRAFTING
    }
}